libpod/container_internal.go

```diff
@@ -1300,6 +1300,9 @@ func (c *Container) start() error {
 
 	c.newContainerEvent(events.Start)
 
+	// Reset any previous errors as we managed to start the container successfully here.
+	c.state.Error = ""
```

Comment:
This feels like it should happen in init() instead, as part of resetting the state from the last run of the container, but I won't argue too loud if you want to keep it here

Reply:
Agree it would fit there, I thought it may best to only do when we now started worked but thinking about it if init or start didn't work we overwrite with the new error anyway so there is no practical difference.